src/javascript/app/pages/user/account/authenticate.js

```diff
@@ -339,31 +374,54 @@ const Authenticate = (() => {
         processFiles(files);
     };
 
+    const showNimcDialog = () =>  new Promise(resolve => {
+        Dialog.confirm({
+            id               : 'nimc_notice_dialog',
+            localized_title  : localize('IMPORTANT!'),
+            localized_message: localize('[_1]Before uploading your NIMC,[_2] [_3]- Ensure that only the first six and last four digits of the PAN are visible on the front.[_4][_3]- Cover the 3-digit CVV code on the back.[_4][_5][_3]Alternatively, you may upload your passport or driving licence.[_4]', ['<strong>', '</strong>','<p>','</p>',nimc_notice_images]),
+            ok_text          : localize('Got it'),
+            cancel_text      : localize('Upload another document'),
+        }).then((response) => resolve(response));
+    });
+
     /**
      * On submit button click
      */
-    const submitFilesUns = ($files) => {
+    const submitFilesUns = async ($files) => {
+        const is_nigeria = country_code === 'NGA';
+
         if ($button_uns.length && $button_uns.find('.barspinner').length) { // it's still in submit process
             return;
         }
+
+        if (is_nigeria && !has_accepted_nimc_notice){
+            await showNimcDialog().then(result => has_accepted_nimc_notice = result);
+
+            if (!has_accepted_nimc_notice){
+                return;
+            }
+        }
```

Comment:
```suggestion
        if (is_nigeria && !has_accepted_nimc_notice && !await showNIMCDialog()){
            return;
        }
```

Reply:
We have to set `has_accepted_nimc_notice` inside the condition, because before we show the dialog we should check if user agreed it before or not.